test/libsolidity/semanticTests/functionCall/return_size_shorter_than_expected.sol

```diff
@@ -0,0 +1,32 @@
+interface LongReturn {
+    function f() external pure returns (uint[20] memory);
+}
+contract ShortReturn {
+    function f() external pure returns (bytes32) {}
+}
+
+contract Test {
+    function test() public returns (uint) {
+        ShortReturn shortReturn = new ShortReturn();
+        uint freeMemoryBefore;
+        assembly {
+            freeMemoryBefore := mload(0x40)
+        }
+
+        LongReturn(address(shortReturn)).f();
+
+        uint freeMemoryAfter;
+
+        assembly {
+            freeMemoryAfter := mload(0x40)
+        }
+
+        return freeMemoryAfter - freeMemoryBefore;
+    }
+}
+// ====
+// EVMVersion: <=homestead
+// compileViaYul: true
+// ----
+// test() -> 0x0500
```

Comment:
I was going to say that there must be something wrong here, because it's allocating 0x500 bytes even though we have the condition to reduce it. Just noticed that it's only for `homestead` though. We should really have coverage for newer EVM on this too. Can you add a variant of this test for `EVMVersion: >homestead` too?